sydent/http/servlets/__init__.py

```diff
@@ -100,6 +109,31 @@ def inner(self, request, *args, **kwargs):
             })
     return inner
 
+def deferjsonwrap(f):
+    def reqDone(resp, request):
+        request.setResponseCode(200)
+        request.write(json.dumps(resp).encode("UTF-8"))
+        request.finish()
+
+    def reqErr(failure, request):
+        if failure.check(MatrixRestError) is not None:
+            request.setResponseCode(failure.value.httpStatus)
+            request.write(json.dumps({'errcode': failure.value.errcode, 'error': failure.value.error}))
+        else:
+            logger.error("Request processing failed: %r, %s", failure, failure.getTraceback())
+            request.setResponseCode(500)
+            request.write(json.dumps({'errcode': 'M_UNKNOWN', 'error': 'Internal Server Error'}))
+        request.finish()
```

Comment:
Can we just make `inner` a `@defer.inlineCallbacks` and then use normal try/except?

Reply:
yeah, possibly, although I faffed around for quite a bit trying various different shapes for this and it ended up being very subtle so I'm inclined to leave it for now as I know this works and it there's nothing to stop us tidying up in the future.